File: 231.2-的幂.cpp
/*
 * @lc app=leetcode.cn id=231 lang=cpp
 *
 * [231] 2的幂
 */

// @lc code=start
class Solution {
public:
    bool isPowerOfTwo(int n) {
        if (n == 0) return false;
        if (n == 1) return true;
        long f = 2;
        for (int i = 0; i < n; i++) {
            if (f == n) {
                return true;
            }
            if (f >n) return false;
            f *= 2;
        }
        return false;
    }
};
// @lc code=end

File: 231.2-的幂.go
/*
 * @lc app=leetcode.cn id=231 lang=golang
 *
 * [231] 2 的幂
 */

// @lc code=start
func isPowerOfTwo(n int) bool {
	if n == 1 {
		return true
	}
	if n == 0 {
		return false
	}
	f := float64(n)
	for {
		r:=f
		f = f / 2.0
		if f == 1 {
			return true
		}
		if f != float64(int(r) / 2) {
			return false
		}
	}
}
// @lc code=end

